src/rpc/processor.zig

```diff
@@ -0,0 +1,668 @@
+        _ = allocator;
+        _ = self;
+        _ = pubkey;
+        _ = config;
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBalance(self: *Self, allocator: std.mem.Allocator, pubkey: t.Pubkey, config: t.RpcContextConfig) t.Result(t.RpcResponse(u64)) {
+        _ = allocator;
+        _ = self;
+        _ = pubkey;
+        _ = config;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlock(self: *Self, allocator: std.mem.Allocator, slot: t.Slot, config: ?t.RpcBlockConfig) t.Result(t.UiConfirmedBlock) {
+        _ = allocator;
+        _ = self;
+        _ = slot;
+        _ = config;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlockCommitment(self: *Self, allocator: std.mem.Allocator, block: t.Slot) t.Result(t.RpcBlockCommitment) {
+        _ = allocator;
+        _ = self;
+        _ = block;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlockHeight(self: *Self, allocator: std.mem.Allocator, config: t.RpcContextConfig) t.Result(u64) {
+        _ = allocator;
+        _ = self;
+        _ = config;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlockProduction(self: *Self, allocator: std.mem.Allocator, config: ?t.RpcBlockProductionConfig) t.Result(t.RpcResponse(t.RpcBlockProduction)) {
+        _ = allocator;
+        _ = self;
+        _ = config;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlocks(self: *Self, allocator: std.mem.Allocator, start_slot: t.Slot, end_slot: t.Slot, commitment: ?t.CommitmentConfig) t.Result([]t.Slot) {
+        _ = allocator;
+        _ = self;
+        _ = start_slot;
+        _ = end_slot;
+        _ = commitment;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlocksWithLimit(self: *Self, allocator: std.mem.Allocator, start_slot: t.Slot, limit: usize, commitment: ?t.CommitmentConfig) t.Result([]t.Slot) {
+        _ = allocator;
+        _ = self;
+        _ = start_slot;
+        _ = limit;
+        _ = commitment;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getBlockTime(self: *Self, allocator: std.mem.Allocator, slot: t.Slot) t.Result(?t.UnixTimestamp) {
+        _ = allocator;
+        _ = self;
+        _ = slot;
+
+        return .{ .Err = t.Error.Unimplemented };
+    }
+
+    pub fn getClusterNodes(
+        self: *Self,
+        allocator: std.mem.Allocator,
+    ) t.Result([]t.RpcContactInfo) {
+        var table_rlock = self.gossip_service.gossip_table_rw.read();
+        defer table_rlock.unlock();
+        var table: *const GossipTable = table_rlock.get();
+
+        var contact_infos = table.getAllContactInfos() catch return .{ .Err = t.Error.Internal };
+        var rpc_contact_infos = std.ArrayList(t.RpcContactInfo).initCapacity(allocator, contact_infos.items.len) catch return .{ .Err = t.Error.Internal };
```

Comment:
will all rpc methods need to return this Internal Error? - catch statements might be excessive on more complex rpc calls